leakcanary-android-instrumentation/src/androidTest/java/leakcanary/Profiler.kt

```diff
@@ -34,6 +37,48 @@ internal object Profiler {
     SharkLog.d { "Sampling stopped! Proceeding..." }
   }
 
+  /**
+   * Executes the given function [block] with CPU sampling via Profiler and returns the result of
+   * the function execution.
+   * First, it awaits for Profiler to be attached at start of sampling, then executes [block]
+   * and finally waits for sampling to stop. See [waitForSamplingStart] and [waitForSamplingStop]
+   * for more details.
+   */
+  fun <T> runWithProfilerSampling(block: () -> T): T {
+    waitForSamplingStart()
+    val result = block()
+    waitForSamplingStop()
+    return result
+  }
+
+  private const val TRACES_FOLDER = "/sdcard/traces/"
+  private const val TRACE_NAME_PATTERN = "yyyy-MM-dd_HH-mm-ss_SSS'.trace'"
+  private const val BUFFER_SIZE = 50 * 1024 * 1024
+  private const val TRACE_INTERVAL_US = 1000
+
+  /**
+   * Executes the given function [block] with method tracing to SD card and returns the result of
+   * the function execution.
+   * Tracing is performed with [Debug.startMethodTracingSampling] which uses sampling with
+   * [TRACE_INTERVAL_US] microseconds interval.
+   * Trace file will be stored in [TRACES_FOLDER] and can be pulled via `adb pull` command.
+   * See Logcat output for an exact command to retrieve trace file
+   */
+  fun <T> runWithMethodTracing(block: () -> T): T {
```

Comment:
nit: wondering if folder, buffer size and interval should be params with default values. Not necessary though

Reply:
Maybe in the future, right now I don't expect us to use any other configurations. It's an easy fix to expand it anyway.